lib/hooks.js

```diff
@@ -119,8 +130,15 @@ var Hooks = {
     if (hookTypes[hookType] && hookTypes[hookType].sync) {
       hooks.forEach(function(hook) {
```

Comment:
It will break the order of execution. Proxied hooks must be called after primary hooks has been completed. Otherwise `beforeUpdate` errors will not stop the `beforeSave` hook